crates/templates/src/template.rs

```diff
@@ -358,6 +358,43 @@ impl Template {
             }
         }
     }
+
+    pub(crate) fn check_compatible_manifest_format(
+        &self,
+        manifest_format: u32,
+    ) -> anyhow::Result<()> {
+        let Some(content_dir) = &self.content_dir else {
+            return Ok(());
+        };
+        let manifest_tpl = content_dir.join("spin.toml");
+        if !manifest_tpl.is_file() {
+            return Ok(());
+        }
+
+        // We can't load the manifest template because it's not valid TOML until
+        // substituted, so GO BIG or at least GO CRUDE.
+        let Ok(manifest_tpl_str) = std::fs::read_to_string(&manifest_tpl) else {
+            return Ok(());
+        };
+        let is_v1_tpl = manifest_tpl_str.contains("spin_manifest_version = \"1\"");
```

Comment:
why do we escape quotes here and not for spin_manifest_version = 2?

Reply:
In v1 there are quotes around the 1.  In v2 there are no quotes around the 2.